src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(newState, action.extraData);
+    }
+
+    if (action.type === 'clear') {
+      for (const key in newState) {
+        delete newState[key];
+      }
+    }
+
+    if (action.type === 'removeProperties') {
+      for (const key in newState) {
+        if (action.keysToRemove.includes(key)) {
+          delete newState[key];
```

Comment:
U can do it easier. For example: 
```
newState = {};
```
and u'll have clear object in your array.

Reply:
Switch added)